eureka2-client/src/test/java/com/netflix/eureka2/client/interest/BatchAwareIndexRegistryTest.java

```diff
@@ -0,0 +1,165 @@
+package com.netflix.eureka2.client.interest;
+
+import com.netflix.eureka2.interests.ChangeNotification;
+import com.netflix.eureka2.interests.ChangeNotification.Kind;
+import com.netflix.eureka2.interests.Index.InitStateHolder;
+import com.netflix.eureka2.interests.IndexRegistry;
+import com.netflix.eureka2.interests.Interest;
+import com.netflix.eureka2.interests.Interests;
+import com.netflix.eureka2.interests.StreamStateNotification;
+import com.netflix.eureka2.registry.instance.InstanceInfo;
+import com.netflix.eureka2.rx.ExtTestSubscriber;
+import com.netflix.eureka2.testkit.data.builder.SampleChangeNotification;
+import org.junit.Before;
+import org.junit.Test;
+import rx.Observable;
+import rx.subjects.PublishSubject;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Tomasz Bak
+ */
+public class BatchAwareIndexRegistryTest {
```

Comment:
We should also add tests for:
- shutdown and cleanup
- handling of onError and onCompleted cases for the batchHintStream
- handling of onError and onCompleted cases for the dataStream